### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by misode**
This refactor adds schemas in a custom format which will allow easy updating the loot table format, as well as supporting multiple versions.
It reduces file size significantly. Fields are no longer hard coded, but are data-driven.

### CODE REVIEW

File: locales/zh-CN.json [hunk 1]
```diff
@@ -244,7 +244,7 @@
 	"condition": "条件",
 	"copy": "复制",
 	"count": "数量",
-	"description": "适用于 Minecraft 1.14 的战利品表生成器",
+	"description": "适用于 Minecraft 的战利品表生成器",
```
Comment (SPGoding) [inline locales/zh-CN.json]: `"description": "Minecraft 战利品表生成器"` is better xD

**Comment by misode** [comment]
I'm thinking of stuff that's missing. Mind checking over this PR? @SPGoding

File: schemas/1.14.json [hunk 2]
```diff
@@ -0,0 +1,1216 @@
...
+            }
+          ]
+        },
+        {
+          "id": "conditions",
+          "type": "array",
+          "translate": "condition",
+          "values": "condition",
+          "button": "header"
+        }
```
Comment (SPGoding) [inline schemas/1.14.json]: Missing `effects` for `minecraft:set_stew_effect`, and `destination` (`目标`), `decoration` (`图标`), `zoom` (`缩放等级`, defaults to `2`), `search_radius` (`搜寻半径`, defaults to `50`) and `skip_existing_chunks` (`跳过已生成区块`, defaults to `true`) for `minecraft:exploration_map`.

File: schemas/1.14.json [hunk 3]
```diff
@@ -0,0 +1,1216 @@
...
+        },
+        {
+          "id": "effects",
+          "type": "map",
+          "translate": "$entity.status_effect",
+          "values": {
+            "type": "object",
+            "value": "status_effect"
+          }
+        }
```
Comment (SPGoding) [inline schemas/1.14.json]: Missing `flags` and `equipment`?

File: locales/zh-CN.json [hunk 4]
```diff
@@ -233,71 +265,27 @@
 	"add_modifier": "添加属性修饰器",
 	"add_operation": "添加操作",
 	"add_pool": "添加随机池",
-	"add_score": "添加分数",
 	"add_term": "添加条件",
```
Comment (SPGoding) [inline locales/zh-CN.json]: `add_foo` should be changed to `foo_add`

File: locales/zh-CN.json [hunk 5]
```diff
@@ -233,71 +265,27 @@
...
-	"remove_block_state": "移除方块状态",
 	"remove_condition": "移除条件",
 	"remove_enchantment": "移除附魔",
 	"remove_entry": "移除项目",
 	"remove_function": "移除函数",
 	"remove_modifier": "移除属性修饰器",
 	"remove_operation": "移除操作",
 	"remove_pool": "移除随机池",
-	"remove_score": "移除分数",
 	"remove_term": "移除条件",
```
Comment (SPGoding) [inline locales/zh-CN.json]: `remove_foo` should be changed to `foo_remove`

File: schemas/1.14.json [hunk 6]
```diff
@@ -0,0 +1,1216 @@
...
+        "translateValue": "$table.$type",
+        "default": "minecraft:generic",
+        "values": [
+          "minecraft:empty",
+          "minecraft:entity",
+          "minecraft:block",
+          "minecraft:chest",
+          "minecraft:fishing",
+          "minecraft:generic"
+        ]
```
[the remark this answers is not in the document]
Comment (misode) [reply schemas/1.14.json]: `advancement_reward` was removed in a later 1.14 snapshot

File: schemas/1.14.json [hunk 7]
```diff
@@ -0,0 +1,1216 @@
...
+      },
+      {
+        "id": "pools",
+        "type": "array",
+        "translate": "pool",
+        "values": "pool"
+      }
+    ]
```
[the remark this answers is not in the document]
Comment (misode) [reply schemas/1.14.json]: I'm not sure about that

File: schemas/1.14.json [hunk 8]
```diff
@@ -0,0 +1,1216 @@
...
+          "button": "header"
+        },
+        {
+          "id": "conditions",
+          "type": "array",
+          "translate": "condition",
+          "values": "condition",
+          "button": "header"
+        }
+      ]
```
[the remark this answers is not in the document]
Comment (misode) [reply schemas/1.14.json]: Rip, how did this happen? 😂

**Comment by misode** [comment]
@SPGoding I believe I made all the changes. It's missing some translations, but since it falls back to English, that's not a huge deal.

**Comment by misode** [comment]
> btw, is there anything I can help to make it support 1.15?

Let's do that in a separate PR. If you want you can start it after I merge this.

**Pull request merged**